scripts/mbedtls_framework/c_parsing_helper.py

```diff
@@ -129,3 +152,17 @@ def read_function_declarations(functions: Dict[str, FunctionInfo],
                                        return_type,
                                        name,
                                        arguments)
+
+_C_TYPEDEF_DECLARATION_RE = r'typedef (?:struct )?(?P<type>\w+) (?P<name>\w+)'
+
+def read_typedefs(filename: str)->  Dict[str, str]:
+    """ Extract type definitions in a {typedef aliased name: original type} dictionary.
+    Multi-line typedef struct are not captured. """
+
+    type_decl = {}
+    with open(filename, encoding='utf-8') as inp:
```

Comment:
I will change that so we can move on, but for the record I think that `read_logical_lines()` is hardly needed.

What it effective is doing is:

1. file.read()
2. re.sub
3. splitlines()
4. enumerate 
5. Iterate through the levels.

What is the method trying to achieve?
- If the ask is to skip comment lines, we can achieve that with rejecting regex, which is why the tool is build and tested for.
- If the ask is to find all matches in a multiline block of text, and return the line number, you can achieve it by calling re.match() to map the indexes and then re.finditer() once to have an iterator giving you the matches, that you can pair with the line number.

Why do we need custom methods for built and tested native python functionality?

Reply:
`read_logical_line` skips C comments. There's no native Python functionality for that.

> If the ask is to skip comment lines, we can achieve that with rejecting regex, which is why the tool is build and tested for.
>
> If the ask is to find all matches in a multiline block of text, and return the line number, you can achieve it by calling re.match() to map the indexes and then re.finditer() once to have an iterator giving you the matches, that you can pair with the line number.

Now you're discussing the implementation of `read_logical_lines`. I think you're outlining a different implementation technique. Rewriting `read_logical_lines` is out of scope here since we don't need any new functionality. Furthermore, your proposed technique seems more difficult than the one that's already there: it's hard to do regexp-based replacements but keep track of line numbers.

Reply:
> read_logical_line skips C comments. There's no native Python functionality for that.

I though regex exists for precicely that use-case?

Something like that should work 

    c_comments_rex = re.compile(r'//.*?$|/\*.*?\*/|\'(?:\\.|[^\\\'])*\'|"(?:\\.|[^\\"])*"', re.DOTALL | re.MULTILINE)

    # Remove comment lines
    content = re.sub(c_comments_rex, "", content)

    # Find typedefs
    for m in  re.finditer(_C_TYPEDEF_DECLARATION_RE, content):
        type_decl[m.group("name")] = m.group("type")

But again I am not exactly clear why `read_logical_lines` is needeed and what requirements it fulfills.  

Anyway this is out of scope so I am adjusting the utility function to use `read_logical_lines` so we can move forward.